src/app/modules/verification/verification.service.ts

```diff
@@ -477,12 +479,21 @@ const sendOtpForField = (
   | InvalidNumberError
   | MailSendError
   | NonVerifiedFieldTypeError
+  | SmsLimitExceededError
+  | OtpRequestError
```

Comment:
Shouldn't `otpRequestError` be swallowed in this method rather than surfaced to caller?

As in, with my (probably very broken) understanding, an SMS limit error should be surfaced because the form is not behaving as a user would expect, but `otpRequestError` means the field didn't need an opt anyway, and so the form is performing correctly if no otp is sent

Or more likely I probably need a bit of help to understand what a field being not verifiable means here?

Reply:
we surface all errors to the controllers anyways, and the controller will have a error handler that returns the correct error code depending on the error

Reply:
Thanks! as what Kar Rui said. the error is surfaced to the controller so that it can be returned and then displayed on the frontend

In this case, the error is triggered when client still has the old form where OTP verification is enabled. the form (from the client's perspective) is not performing correctly if no otp is sent, hence the need for error message

Reply:
Thanks guys! Just for extra info for me: what does `isVerifiable` on the field represent?

Reply:
Whether the Mobile Number field has SMS verification enabled or not:

![Screenshot 2021-08-19 at 9 59 23 AM](https://user-images.githubusercontent.com/22133008/129995632-c839fa48-6059-4cdb-a062-77c23122c7ca.png)
![Screenshot 2021-08-19 at 9 59 27 AM](https://user-images.githubusercontent.com/22133008/129995639-efcaccee-09ab-446d-babd-9924d50eecba.png)

Reply:
Hmm, but so why is there an error here?

Reply:
as explained together with karrui over huddle :)

Reply:
Understood the flow. I had the complete incorrect assumption of the starting point 😑😰.